tools/toolchain/scripts/install_openblas.sh

```diff
@@ -45,7 +45,7 @@ case "$with_openblas" in
             cd OpenBLAS-${openblas_ver}
 
             for patch in "${PATCHES[@]}" ; do
-                patch -p1 < "${SCRIPT_DIR}/${patch}"
+                patch -p1 < "${SCRIPT_DIR}/${patch}" || [ $(getenforce) = 'Enforcing' ] && true || false
```

Comment:
why do we have to hide errors from `patch`?

Reply:
The patch command completes with an error when selinux is Enforcing and the toolchain run stops, since the context of the patched files could not be changed. Do you know any better solution for that issue?

Reply:
This is strange. Users' files are usually have the 'unconfined' context, so they are exempt from SELinux rules. Besides, non-SELinux distributives lack the 'getenforce' command.

Reply:
Do you still have the original error message?
I guess the way distros do it is to make sure that the files have the proper permissions/contexts/ownership after extracting such that when applying a patch no context change should occur.

Reply:
The patch command
```
patch -p1 < /data/user/krack/github/mkrack/cp2k/tools/toolchain/scripts/openblas-0.3.6-disable-avx512.patch
```
causes the error message
```
patching file kernel/x86_64/KERNEL.SKYLAKEX
patch: setting attribute security.selinux for security.selinux: Permission denied
```

Reply:
@mkrack can you check whether running `restorecon -rv tools/toolchain/` prior to `install_cp2k_toolchain.sh` and/or running it explicitly in the OpenBLAS script after unpacking the tarball changes the behaviour?

The proposed patch needs some work, I get `getenforce: command not found`. Maybe combining with `[ -f /sys/fs/selinux ]` might be a way around, unless you want to explicitly check the command first using `command -v getenforce > /dev/null`

Reply:
restorecon returns also a permission denied error
```
restorecon reset /data/user/krack/github/mkrack/cp2k/tools/toolchain/scripts context unconfined_u:object_r:unlabeled_t:s0->unconfined_u:object_r:default_t:s0
restorecon set context /data/user/krack/github/mkrack/cp2k/tools/toolchain/scripts->unconfined_u:object_r:default_t:s0 failed:'Permission denied'
```
and thus does not help.

Reply:
It seems that this a problem of the 'patch' program itself, that tries to modify the protected 'security.selinux' extended attribute directly using setxattr(2). The current method to fix it is to recompile patch using the '--disable-xattr' configure option. By the way, setxattr(2) syscall fails on the 'security.selinux' attribute even when SELinux operates in permissive mode.

Just curiosity, which Linux distribution do you use (the latest Fedora?). The patch program in CentOS 7 seems to be unaffected.

Reply:
@schulkov thanks for the explanation.
The cluster node is running the Red Hat Enterprise Linux Server release 7.6 (Maipo).

Reply:
@dev-zero I added the explicit check for the availability of the getenforce command as suggested, though I do not understand why you get an error on systems without getenforce, since that branch with getenforce is only processed when the patch command fails even in the prior version.

Reply:
@mkrack thanks. I should have to be more precise: I get the cited error if the patch command fails, but given that in the future people might not know about it this whole thing anymore it is a good idea to not add unrelated error output on top...

Reply:
@mkrack it might be a good idea to open an issue with either RedHat (or your IT dept?). I have the impression that you're missing an SELinux rule allowing a user a context switch or that the `/data` directory is incorrectly tagged: I've tried reproducing the error with a CentOS 7 with various combinations of setting `chcon -t unlabeled_t` on either the patch, the target directory or the base directory (with a default of `default_t`) where I build CP2K, to no avail. But I think that for some reason you're ending up with a mix of contexts (they're usually only preserved, not explicitly set) without the permission to change between them...
Btw, what do you get when you do `ls -lisaZ tools/toolchain/scripts tools/toolchain/build`?

Reply:
All files and folders in `tools/toolchain/scripts` and `tools/toolchain/build` have the context
```
unconfined_u:object_r:unlabeled_t:s0
```
I will check with our Admins.